core/common/src/test/java/alluxio/util/ShellUtilsTest.java

```diff
@@ -110,4 +134,66 @@ public void getMountInfo() throws Exception {
     List<UnixMountInfo> info = ShellUtils.getUnixMountInfo();
     assertTrue(info.size() > 0);
   }
+
+  @Test
+  public void execCommandTolerateFailure() throws Exception {
+    // create temp file
+    File testDir = AlluxioTestDirectory.createTemporaryDirectory("command");
+
+    File testFile = createFileInDir(testDir, "testFile");
+
+    // ls temp file
+    String[] testCommandSucceed = new String[]{"ls",
+            String.format("%s", testDir.getAbsolutePath())};
+    ShellUtils.CommandReturn crs = ShellUtils.execCommandTolerateFailure(testCommandSucceed);
+    assertEquals(0, crs.getExitCode());
+    assertTrue(crs.getStdOut().contains(testFile.getName()));
+
+    // do sth wrong
+    String[] testCommandFail = new String[]{"ls",
+            String.format("%saaaa", testDir.getAbsolutePath())};
+    ShellUtils.CommandReturn crf = ShellUtils.execCommandTolerateFailure(testCommandFail);
+    System.out.println(crf.getFormattedOutput());
+    assertFalse(crf.getExitCode() == 0);
+
+    // if there's no such command there will be IOException
+    mExceptionRule.expect(IOException.class);
+    mExceptionRule.expectMessage("No such file or directory");
+    String[] testCommandExcept = new String[]{"lsa",
+            String.format("%s", testDir.getAbsolutePath())};
+    // lsa is not a valid executable
+    ShellUtils.execCommandTolerateFailure(testCommandExcept);
+  }
+
+  @Test
+  public void sshExecuteCommandTolerateFailure() throws Exception {
```

Comment:
It's worth debating how/whether we want to test SSH. Currently this relies on password-less loopback access.

Reply:
Seems Jenkins doesn't have password-less SSH to localhost. So either we manual test SSH, or bring in a 3rd part implementation of invoking SSH calls.